### PR TITLE
unique users with email

### PR DESCRIPTION
**Pull request by lucaventrice**
create model manager and user model with email as the unique identifier
create new db migrations that uses custom user model

### CODE REVIEW

**Comment by lucaventrice** [comment]
Beim ausführen bekam ich folgende Fehlermeldung: 
`~/dev/private/django/todo_drf(unique-users*) » python manage.py makemigrations && migrate                  1 ↵ luca@lucas-mbp
Traceback (most recent call last):
  File "/Users/luca/dev/private/django/todo_drf/manage.py", line 22, in <module>
    main()
  File "/Users/luca/dev/private/django/todo_drf/manage.py", line 18, in main
    execute_from_command_line(sys.argv)
  File "/Users/luca/.pyenv/versions/3.9.1/lib/python3.9/site-packages/django/core/management/__init__.py", line 401, in execute_from_command_line
    utility.execute()
  File "/Users/luca/.pyenv/versions/3.9.1/lib/python3.9/site-packages/django/core/management/__init__.py", line 395, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/Users/luca/.pyenv/versions/3.9.1/lib/python3.9/site-packages/django/core/management/base.py", line 330, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/Users/luca/.pyenv/versions/3.9.1/lib/python3.9/site-packages/django/core/management/base.py", line 368, in execute
    self.check()
  File "/Users/luca/.pyenv/versions/3.9.1/lib/python3.9/site-packages/django/core/management/base.py", line 392, in check
    all_issues = checks.run_checks(
  File "/Users/luca/.pyenv/versions/3.9.1/lib/python3.9/site-packages/django/core/checks/registry.py", line 70, in run_checks
    new_errors = check(app_configs=app_configs, databases=databases)
  File "/Users/luca/.pyenv/versions/3.9.1/lib/python3.9/site-packages/django/core/checks/urls.py", line 13, in check_url_config
    return check_resolver(resolver)
  File "/Users/luca/.pyenv/versions/3.9.1/lib/python3.9/site-packages/django/core/checks/urls.py", line 23, in check_resolver
    return check_method()
  File "/Users/luca/.pyenv/versions/3.9.1/lib/python3.9/site-packages/django/urls/resolvers.py", line 408, in check
    for pattern in self.url_patterns:
  File "/Users/luca/.pyenv/versions/3.9.1/lib/python3.9/site-packages/django/utils/functional.py", line 48, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/Users/luca/.pyenv/versions/3.9.1/lib/python3.9/site-packages/django/urls/resolvers.py", line 589, in url_patterns
    patterns = getattr(self.urlconf_module, "urlpatterns", self.urlconf_module)
  File "/Users/luca/.pyenv/versions/3.9.1/lib/python3.9/site-packages/django/utils/functional.py", line 48, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/Users/luca/.pyenv/versions/3.9.1/lib/python3.9/site-packages/django/urls/resolvers.py", line 582, in urlconf_module
    return import_module(self.urlconf_name)
  File "/Users/luca/.pyenv/versions/3.9.1/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 790, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/Users/luca/dev/private/django/todo_drf/todo_drf/urls.py", line 21, in <module>
    from users import views as user_views
  File "/Users/luca/dev/private/django/todo_drf/users/views.py", line 5, in <module>
    from .forms import SignUpForm
  File "/Users/luca/dev/private/django/todo_drf/users/forms.py", line 6, in <module>
    class SignUpForm(UserCreationForm):
  File "/Users/luca/.pyenv/versions/3.9.1/lib/python3.9/site-packages/django/forms/models.py", line 253, in __new__
    fields = fields_for_model(
  File "/Users/luca/.pyenv/versions/3.9.1/lib/python3.9/site-packages/django/forms/models.py", line 142, in fields_for_model
    opts = model._meta
AttributeError: 'str' object has no attribute '_meta'`
Weiss du an was das liegen könnte?

**Comment by lucaventrice** [comment]
@mbaechtold requesting review.

File: api/migrations/0002_task_user.py [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
 
+import django.db.models.deletion
 from django.conf import settings
 from django.db import migrations
 from django.db import models
 
 
 class Migration(migrations.Migration):
 
+    initial = True
```
Comment (mbaechtold) [inline api/migrations/0002_task_user.py:11]: Die Migration `api/migrations/0001_initial.py` ist ja auch eine "initial" Migration, jetzt haben wir zwei initiale Migrationen. Das scheint Django aber nicht zu stören 😄 Ich würde das trotzdem entfernen.

File: users/admin.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+from django.contrib.auth.admin import UserAdmin
+
+from .models import CustomUser
+
+
+class CustomUserAdmin(UserAdmin):
+    model = CustomUser
```
Comment (mbaechtold) [inline users/admin.py:7]: ✂️ Das hat m.E. keinen Effekt.

**Review by mbaechtold — CHANGES_REQUESTED** [review]
Das schaut schon sehr gut aus. 

Allerdings kann ich im Django Admin noch keine Benutzer anlegen. Du musst dein Custom User Model noch im Admin registrieren (siehe https://docs.djangoproject.com/en/3.1/ref/contrib/admin/).

Bitte stelle dann auch gleich sicher, dass man Vor- und Nachname des Benutzers im Django Admin bearbeiten kann.

**Review by mbaechtold — APPROVED** [review]

**Pull request merged**